src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,143 @@
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      // TODO: Display loaded logs in table view
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  async saveLogs(logs) {
+    try {
+      await save.saveAsJSON(logs);
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  renderSideBar(logIndex) {
+    const logStringfy = JSON.stringify(this.logs[logIndex]);
+    this.showLogDetails.textContent = logStringfy;
+    this.showLogDetailWrapper.style.display = 'block';
+    this.logTableWrapper.classList.add('width-60');
+  }
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      const selectedRowIndex = event.target.closest('tr').attributes['index']
+        .value;
+
+      if (selectedRowIndex == null) {
+        return;
+      }
+
+      this.renderSideBar(selectedRowIndex);
+    } else {
+      throw new Error('wrong event type found ' + event.type);
+    }
+  }
+
+  renderNewRow(logs) {
+    for (const log of logs) {
+      const newRow = this.table.insertRow(1);
+      newRow.insertCell(0).textContent = log.id;
+      newRow.insertCell(1).textContent = log.timeStamp;
+      newRow.insertCell(2).textContent = log.type;
+      newRow.insertCell(3).textContent = log.name;
+      newRow.insertCell(4).textContent = log.viewType || 'undefined';
+      newRow.setAttribute('index', this.rowIndex++);
+
+      newRow.addEventListener('click', this);
+      this.logs.push(log);
+    }
+  }
+
+  async init() {
+    const existingLogs = await this.getExistingLogs();
+
+    if (existingLogs.length) {
+      this.renderNewRow(existingLogs);
+    }
+
+    this.saveLogBtn.addEventListener('click', () => {
+      this.saveLogs(this.logs);
+    });
+
+    this.logInputFile.addEventListener('change', () => {
+      this.loadLogBtn.removeAttribute('disabled');
+    });
+
+    this.loadLogBtn.addEventListener('click', () => {
+      if (this.logInputFile.files.length > 0) {
+        this.loadFile(this.logInputFile.files[0]);
+        this.logInputFile.value = '';
+        this.loadLogBtn.setAttribute('disabled', true);
+      }
+    });
+
+    this.closeBtn.addEventListener('click', () => {
+      if (this.showLogDetailWrapper.style.display === 'block') {
+        this.showLogDetailWrapper.style.display = 'none';
+        this.logTableWrapper.classList.remove('width-60');
+      }
+    });
```

Comment:
All these event handlers could be merged into the `handleEvent` method in this class, and then inside there we may decide if the event was expected and how we should handle it, e.g. and handleEvent would be looking more or less like:
```js
    if (event.type === 'click') {
      const logDetails = event.target.closest('tr')?._log;

      if (logDetails) {
        // Call the method to open the log details in the sidebar.
        return;
      }

      switch (event.target) {
      case this.saveLogBtn:
        // Call the method that will save the logs to file
        break; 
      case this.closeBtn:
        // Call the method that will close the sidebar.
        break;
      default:
        // Throw error for the click event on an unexpected element.
      }
    } else if (event.type === 'change' && event.target == this.logInputFile) {
      // Load the file..      
    } else {
      // throw an error for an unexpected event type
    }
```

Reply:
Thanks for noticing. It's done now :-)